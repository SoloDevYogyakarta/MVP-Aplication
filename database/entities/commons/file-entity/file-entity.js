"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fileEntity = exports.FileEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class FileEntity extends base_entity_1.BaseEntity {
}
exports.FileEntity = FileEntity;
exports.fileEntity = entity_1.sequelize.define('FILE', {
    public_id: sequelize_1.DataTypes.STRING,
    filename: sequelize_1.DataTypes.STRING,
    originalname: sequelize_1.DataTypes.STRING,
    filepath: sequelize_1.DataTypes.STRING,
    type: sequelize_1.DataTypes.STRING,
    width: sequelize_1.DataTypes.INTEGER,
    height: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'FILE',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=file-entity.js.map