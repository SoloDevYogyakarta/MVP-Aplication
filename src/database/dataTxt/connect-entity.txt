{"id":895,"source_id":"k8TJ_2ROjCwho9XJlfZRl","foreign_id":"Dk1u1tKFZvPDubveGZhUc","public_id":"8kOUYHeXhm9NpFeAcczOI"}