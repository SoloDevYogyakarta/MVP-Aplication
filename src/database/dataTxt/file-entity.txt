{"id":2290,"public_id":"7pYVAFeclZYY6wnqTvHFQ","filename":null,"originalname":null,"filepath":null,"type":null}