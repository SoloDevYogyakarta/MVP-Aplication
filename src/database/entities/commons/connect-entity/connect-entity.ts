import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ConnectEntity extends BaseEntity {
  source_id: string;
  foreign_id: string;
}

export type ConnectCreationAttribute = Optional<ConnectEntity, 'id'>;
export interface ConnectInstance
  extends Model<ConnectCreationAttribute, ConnectEntity>,
    ConnectEntity {}
export const connectEntity = sequelize.define<ConnectInstance>(
  'CONNECT',
  {
    public_id: DataTypes.STRING,
    source_id: DataTypes.STRING,
    foreign_id: DataTypes.STRING,
  } as ModelAttributes<ConnectInstance>,
  {
    tableName: 'CONNECT',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
