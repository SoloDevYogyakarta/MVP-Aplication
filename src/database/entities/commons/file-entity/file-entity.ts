import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class FileEntity extends BaseEntity {
  filename: string;
  filepath: string;
  type: string;
}

export type FileCreationAttribute = Optional<FileEntity, 'id'>;
export interface FileInstance
  extends Model<FileCreationAttribute, FileEntity>,
    FileEntity {}
export const fileEntity = sequelize.define<FileInstance>(
  'FILE',
  {
    public_id: DataTypes.STRING,
    filename: DataTypes.STRING,
    filepath: DataTypes.STRING,
    type: DataTypes.STRING,
  } as ModelAttributes<FileInstance>,
  {
    tableName: 'FILE',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
