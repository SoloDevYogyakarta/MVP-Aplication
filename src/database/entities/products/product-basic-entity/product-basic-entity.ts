import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ProductBasicEntity extends BaseEntity {
  mechanis_name: string;
  desc: string;
}

export type ProductBasicCreationAttribute = Optional<ProductBasicEntity, 'id'>;
export interface ProductBasicInstance
  extends Model<ProductBasicCreationAttribute, ProductBasicEntity>,
    ProductBasicEntity {}
export const productBasicEntity = sequelize.define<ProductBasicInstance>(
  'BASIC',
  {
    public_id: DataTypes.STRING,
    mechanis_name: DataTypes.STRING,
    desc: DataTypes.STRING,
  } as ModelAttributes<ProductBasicInstance>,
  {
    tableName: 'BASIC',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
