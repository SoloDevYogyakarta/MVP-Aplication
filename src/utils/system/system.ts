import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
