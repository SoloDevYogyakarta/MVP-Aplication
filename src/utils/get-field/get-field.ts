import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
