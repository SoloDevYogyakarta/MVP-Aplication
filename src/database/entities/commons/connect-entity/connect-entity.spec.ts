import { ModelCtor } from 'sequelize';
import { getField } from '../../../../utils/get-field/get-field';
import { createpath } from '../../../../utils/system/system';
import { productBasicEntity } from '../../products/product-basic-entity/product-basic-entity';
import { fileEntity } from '../file-entity/file-entity';
import { connectEntity, ConnectInstance } from './connect-entity';

describe('connectEntity', () => {
  let entity: ModelCtor<ConnectInstance>;

  beforeEach(() => {
    entity = connectEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  it('create new', async () => {
    const { public_id } = getField('basic-entity');
    const findOne = await productBasicEntity.findOne({ where: { public_id } });
    const arr = Array.from(Array(5).keys());
    for (const empty of arr) {
      const file = await fileEntity.create({});
      file.save();
      const connect = await entity.create({
        source_id: findOne.public_id,
        foreign_id: file.public_id,
      });
      createpath(`../../database/dataTxt/connect-entity.txt`, connect);
    }
  });

  try {
    it('findOne', async () => {
      const { public_id } = getField('connect-entity');
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
