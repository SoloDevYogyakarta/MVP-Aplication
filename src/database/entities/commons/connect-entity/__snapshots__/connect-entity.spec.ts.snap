// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectEntity render correctly 1`] = `[Function]`;
